eth_defi/gmx/core/available_liquidity.py

```diff
@@ -0,0 +1,450 @@
+                short_liquidity = short_max_reserved_usd / short_precision - float(reserved_short)
+
+                # Special handling for single side markets
+                if "2" in market_symbol:
+                    short_pool_amount = short_pool_amount / 2
+                    short_max_reserved_tokens = short_pool_amount * short_reserve_factor
+                    short_max_reserved_usd = short_max_reserved_tokens / short_precision * token_price
+                    short_liquidity = short_max_reserved_usd - float(reserved_short)
+
+                self.log.info(f"Available Short Liquidity: ${short_liquidity:,.2f}")
+
+                # Store results
+                self.output["long"][market_symbol] = long_liquidity
+                self.output["short"][market_symbol] = short_liquidity
+
+            except Exception as e:
+                self.log.error(f"Failed to process market {market_symbol}: {e}")
+                continue
+
+        self.output["parameter"] = "available_liquidity"
+        return self.output
+
+    def _get_data_processing_multicall(self) -> PositionSideData:
+        """Generate the dictionary of available liquidity using efficient multicall batching.
+
+        :returns: Dictionary of available liquidity data with structure:
+            {
+                "long": {market_symbol: liquidity_value, ...},
+                "short": {market_symbol: liquidity_value, ...},
+                "parameter": "available_liquidity"
+            }
+        :rtype: dict
+        """
+        self.log.debug("GMX v2 Available Liquidity using Multicall")
+
+        # Get open interest data
+        open_interest = GetOpenInterest(self.config).get_data()
+
+        # Generate all multicall requests
+        self.log.debug("Generating multicall requests...")
+        encoded_calls = list(self.generate_all_multicalls())
+        self.log.debug(f"Generated {len(encoded_calls)} multicall requests")
+
+        # Create Web3Factory for multicall execution
+        web3_factory = TunedWeb3Factory(rpc_config_line=self.config.web3.provider.endpoint_uri)
+
+        # Execute all multicalls efficiently
+        self.log.debug("Executing multicalls...")
+        multicall_results: dict[str, dict[str, EncodedCallResult]] = defaultdict(dict)
+
+        for call_result in read_multicall_chunked(
+            chain_id=self.config.web3.eth.chain_id,
+            web3factory=web3_factory,
+            calls=encoded_calls,
+            block_identifier="latest",
+            progress_bar_desc="Loading GMX liquidity data",
+            max_workers=5,  # TODO: Make it dynamic
+        ):
+            market_key = call_result.call.extra_data["market_key"]
+            data_type = call_result.call.extra_data["data_type"]
+            multicall_results[market_key][data_type] = call_result
+
+        self.log.debug(f"Processed multicalls for {len(multicall_results)} markets")
```

Comment:
Use standard Python `logger = logging.getLogger(__name__)` modle-level logging pattern, like every other module in `eth_defi` package.